http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -164,13 +164,10 @@ protected String getRedirectUri(String state) {
                 url = uriBuilder.build().toString();
             }
 
-            List<String> forwardableQueryParams = Arrays.asList(LOGIN_HINT, DOMAIN_HINT, KC_IDP_HINT, PROMPT, MAX_AGE, UI_LOCALES, SCOPE);
+            List<String> forwardableQueryParams = Arrays.asList(LOGIN_HINT, DOMAIN_HINT, KC_IDP_HINT, PROMPT, MAX_AGE, UI_LOCALES);
             List<NameValuePair> forwardedQueryParams = new ArrayList<>(forwardableQueryParams.size());
             for (String paramName : forwardableQueryParams) {
                 String paramValue = getQueryParamValue(facade, paramName);
-                if (SCOPE.equals(paramName)) {
-                    paramValue = addOidcScopeIfNeeded(paramValue);
-                }
```

Comment:
We need to be a bit careful here. This code checks for a scope value in the query params. If the query params doesn't include a value for scope, the resulting `paramValue` for `scope` will be just `openid`. If the query params do include a non-empty value for scope, it will be a space separated list and we will add `openid` to this list if not already present.

Now that we are also allowing the scope to be specified in the OIDC client configuration, we should be combining that with the value present in the query params.

We shouldn't be removing this code.

Reply:
I think this comment still needs to be addressed. The paramValue used to contain any scopes that were specified as a query parameter so we still need to take that into account too.

Reply:
Thanks for pointing this out. 
When I had `paramValue = addOidcScopeIfNeeded(paramValue);`, the scope values specified were not getting added properly. which is why I switched to `deployment.getScope()`. 
So, should I be adding `deployment.getScope()` to paramvalue instead of making it equal to `deployment.getScope()`?

Reply:
I think we can create a set that contains any scopes from paramValue combined with any scopes from deployment.getScope().